# Anti-Vaxx: How I learned to stop getting vaccinated and trust my gut

Chiara Eisner, Joasia Popowicz, Matt Albasi, Sofia Jeremias, Samara Ahmed

## What's this anti-vaccination thing about?
There is a measles outbreak happening in the U.S. right now - 838 confirmed cases, according to the Centers for Disease Control and Prevention. It’s an extremely contagious disease, that can remain in the air for up to two hours someone infected has left. And it’s become a public health nightmare.

The disease starts with a rash at the head and then moves down the body. And it has significant long-term effects. It can cause brain damage, or a loss of hearing or eyesight, sometimes even death. About 1 in 4 measles patients will end up in a hospital.

In 2000, the U.S. declared measles was eliminated. But in 2014, the number of measles cases started to skyrocket, in large part due to anti-vax misinformation spread on social media. 

But groups being opposed to vaccinations in the U.S. is nothing new. The first one, the Anti-Vaccination Society of America, was founded in 1879. And there are court battles against mandated vaccinations that can be traced back to 1902. The modern anti-vax movement actually starts in 1998, after a paper published by Andrew Wakefield, a British doctor who linked the measles, mumps and rubella vaccine to autism. Though Wakefield had his medical license revoked and the paper has since been retracted, the damage was already been done. 

Over the past few years, social media platforms like YouTube and Facebook allowed anti-vax groups to organize in a grassroots way.

## Interview with Renée DiResta

Joasia and Chiara spoke with Renée DiResta on April 22, 2019

Spoke about various objects of focus of anti-vaxxers over the years. “I think at some point research into the Marisol on the preservatives becomes a topic of focus. It's a type of type of mercury.”

Revealed that lately, vaccine-focused churches have been popping up so that families can claim “religious” reasons for anti vaccination of their children.

On YouTube as a platform for spreading misinformation:

“How can we use YouTube to to get our content? I mean, most mainstream media properties won't run their stuff because they know that it's, you know, kind of a debunked conspiracy theory. And so, even though the media will occasionally do some kind of like false balance ISIS, for the most part, they won't, you know, put them on the mainstream news and give them a huge platform so YouTube was the way for them to get their popcorn.”

“They do a good job of influencer marketing.”

“It's just performative, especially in the days of social media, you can make any kind of ridiculous micro outrage micro aggressions go viral.”

Russia has been “an interesting case for a long time” even though the narratives are different from the American ones. Interestingly, the Russian Orthodox communities in the US are very often vaccine hold outs. Lots of YouTube presence, celebrity amplified.

“There's a computer science researcher who's anti vaccine named Stephanie sent us out of MIT, who did this really weird studies saying that it's glyphosate from Monsanto, that's a trace element in the vaccines. Again, just continuing with this like ingredients, ingredients, ingredients kind of thing.”

Sent us an ethnography on background.

Informed us about Washington events: “They were fundraising on GoFundMe, they raise $10,000 to run ads against the Washington bill claiming that it was going to be the start of government forced vaccination of your children.”

Spoke to us about parental motivations: “Autism is just one thing. So there are some, some parents are motivated by this notion of like, purity.”

Not always conservatives, libertarians are often anti-vcxx too: “That becomes a typical libertarian talking point is much more like a parental freedom, parental rights, particularly when the religion element does come into it again, even though no major world religion, or Canada, even the smaller ones that aren't like, you know, fake is opposed to vaccination.”

## What we found on Facebook

### Toxicity

According to Diresta a lot of information has been spread claiming that vaccines contain toxic chemicals

<img src=https://raw.githubusercontent.com/sofiajeremias/Measles/master/Toxcicity.png width=400>
<br>

### Safety

Vaccines are considered unsafe


<img src=https://raw.githubusercontent.com/sofiajeremias/Measles/master/Vaxkidshealthy.png width=400>
<br>

### Liberty

Mandatory vaccines are considered an assault of liberty


<img src=https://raw.githubusercontent.com/sofiajeremias/Measles/master/Screen%20Shot%202019-04-19%20at%2011.20.06%20AM.png width=400>
<br>

    
### Religiosity

Claims that vaccines are created using animal and human parts violate religious beliefs

<img src=https://raw.githubusercontent.com/sofiajeremias/Measles/master/Screen%20Shot%202019-04-19%20at%2010.48.46%20AM.png width=400>
<br>

There are many conspiracies alleging doctors have been killed, or are being paid out by companies like Merk. 

### Conspiracy

<img src=https://raw.githubusercontent.com/sofiajeremias/Measles/master/Screen%20Shot%202019-04-19%20at%202.28.17%20PM.png width=400>
<br>

In [ ]:
%%sh
pip install tldextract

In [ ]:
%%sh
pip install goose3

First, Sofia manually collected the links to articles being shared through a variety of closed Facebook groups.Including:
HOLISTIC LIVES MATTER (Erin Elizabeth from Health Nut News)
Vaccine Talk: A Forum for Both Pro and Anti Vaxers
A.V.P. (Anti-vax Parents)
March Against Monsanto Official Open Group
Revolution For Vaccine Choice
Vaccine Education Network : Natural Health Anti-Vaxx Community
Vaccines Facts and Information

From these groups Sofia made a list of about 30 websites that were frequently linked out to. We wanted to find out what the primary sources were for these articles so we built a webscrapper to do this:

In [ ]:
import csv
from goose3 import Goose
import tldextract

g = Goose()
links = []
with open('sites.csv', "r") as csvfile:
    rows = csv.reader(csvfile)

    for row in rows:
        url = row[0]
        print("Downloading " + url)
        article = g.extract(url=url)
        for link in article.links:
            links.append({
                'link': link,
                'source': url
            })

domains = {}
for link in links:
    domain = tldextract.extract(link['link']).domain
    if domain in domains:
        domains[domain]['count'] += 1
        domains[domain]['links'].append(link)
    else:
        domains[domain] = {
            'count': 1,
            'links': [link]
        }

with open('domains.csv','w') as output:
    writer = csv.writer(output)
    for k,v in domains.items():
        writer.writerow([k,v['count']])

with open('urls.csv','w') as output:
    writer = csv.writer(output)
    for k,v in domains.items():
        for link in v['links']:
            writer.writerow([k,link['link'],link['source']])

This created two files: one with the actual links to articles and another with just the domain names so we could quickly see which sites were being used frequently. 

In [4]:
import pandas as pd
columns = ["domain","frequency"]
domains = pd.read_csv('domains.csv',names=columns)

In [5]:
domains.head()

domain  frequency
0          state          2
1          wired          1
2         eepurl          1
3  specificfeeds          1
4      pinterest          3

In [58]:
domains.sort_values(by=['frequency'],ascending=False)

domain  frequency
5                             NaN        260
34                            nih         83
41                            cdc         50
161                      blogspot         29
94                           nvic         20
128                   naturalnews         20
20                       facebook         15
147             healthytraditions         15
240              thinklovehealthy         15
32                  sciencedirect         13
83                  vaccineimpact         13
6                             bmj         12
18          thefreethoughtproject         10
261                 circleofmamas         10
101                 healthnutnews          9
125                   wakingtimes          9
64                         google          9
93                            who          9
43                          wiley          8
160                       nytimes          7
72       stopmandatoryvaccination          7
195            thevaccinereaction          7
213                           nap          7
154                      addtoany          6
170                        harvoa          6
87                         amazon          6
143                  trueactivist          6
105        childrenshealthdefense          6
86                        youtube          6
29                           plos          5
..                            ...        ...
185                townsendletter          1
42                    jem-journal          1
187                    soundcloud          1
84                 vaccine-injury          1
189                      vaccines          1
46   journalofpsychiatricresearch          1
165                       blogger          1
141                        tumblr          1
153                           fda          1
60                     eurekalert          1
1                           wired          1
58                         utexas          1
146              jennifermargulis          1
56                      nutrition          1
149              informedchoicewa          1
150                      syracuse          1
55             jneuroinflammation          1
152                    mctlawyers          1
53                  aspetjournals          1
164                     juanpagan          1
52              yourlocalguardian          1
156          catherinejfrompovich          1
51                  biomedcentral          1
50                     ghpjournal          1
49                    ahajournals          1
48          autonomicneuroscience          1
47                      thelancet          1
162                 brooklyneagle          1
163                          amny          1
287               newamericannews          1

[288 rows x 2 columns]

We see that after faulty links, the National Institutes of Health (NIH)is the most linked out to domain. After that is the CDC, then personal blogs, and then the National Vaccine Information Center (NVIC), a prominent anti-vaccination site. 

In [98]:
import pandas as pd
columns2 = ["link","url"]
urls = pd.read_csv('urls.csv',names=["site", "link", "source"])

urls

site  \
0                        site   
1                       state   
2                       state   
3                       wired   
4                      eepurl   
5               specificfeeds   
6                   pinterest   
7                   pinterest   
8                   pinterest   
9                         NaN   
10                        NaN   
11                        NaN   
12                        NaN   
13                        NaN   
14                        NaN   
15                        NaN   
16                        NaN   
17                        NaN   
18                        NaN   
19                        NaN   
20                        NaN   
21                        NaN   
22                        NaN   
23                        NaN   
24                        NaN   
25                        NaN   
26                        NaN   
27                        NaN   
28                        NaN   
29                        NaN   
...                       ...   
1032                      aap   
1033                      aap   
1034                     nejm   
1035                     nejm   
1036          semanticscholar   
1037            howstuffworks   
1038         psychiatryonline   
1039                    brown   
1040               bellybelly   
1041         domesticgeekgirl   
1042                parenting   
1043                       nd   
1044  breastfeedingtoday-llli   
1045                       ms   
1046         commonwealthfund   
1047             qualitywatch   
1048                youcaring   
1049                 vactruth   
1050                 vactruth   
1051                 vactruth   
1052                  paraven   
1053                instagram   
1054                instagram   
1055      thevaccinefreechild   
1056      thevaccinefreechild   
1057                   oatext   
1058                   oatext   
1059                   oatext   
1060           vaxxedthemovie   
1061          newamericannews   

                                                   link  \
0                                                  link   
1     https://olis.leg.state.or.us/liz/2019R1/Measur...   
2     https://www26.state.nj.us/doh-shad/indicator/c...   
3     https://www.wired.com/story/facebook-anti-vacc...   
4                              http://eepurl.com/cErfpD   
5     http://www.specificfeeds.com/widgets/emailSubs...   
6     https://www.pinterest.com/pin/create/button/?u...   
7     http://pinterest.com/pin/create/bookmarklet/?m...   
8                 https://www.pinterest.com/naturalnews   
9                                                #_edn1   
10                                               #_edn2   
11                                               #_edn3   
12                                               #_edn4   
13                                               #_edn5   
14                                               #_edn1   
15                                               #_edn1   
16                                               #_edn1   
17                                               #_edn6   
18                                               #_edn6   
19                                               #_edn7   
20                                               #_edn7   
21                                               #_edn7   
22                                            #_ednref1   
23                                            #_ednref2   
24                                            #_ednref3   
25                                            #_ednref4   
26                                            #_ednref5   
27                                            #_ednref6   
28                                            #_ednref7   
29                                                    #   
...                                                 ...   
1032  https://www.aap.org/en-us/advocacy-and-policy/...   
1033  https://www.aap.org/en-us

In [61]:
#taking all the unique values and how many times they occur
urls['domain'].value_counts().shape

(938,)

Unfortunately, the low percentage of overlap meant we couldn't really determine whether or not some studies were frequently being linked out to.

So Sofia and Matt took a look at some of the studies being referenced

"Prevalence and neonatal factors associated with autism spectrum disorders in preterm infants" was one such study. It concluded that shorter periods of gestation in the womb lead to higher causes of autism. Another study was related to serotonin and its relation to IBS, not at all relevant to vaccinations. Manually looking at a sample of the articles sited a few things were prevalent:
Many were very old studies
Most were not related to vaccines but instead to either SIDS, gastrointenstinal health, or other pre-natal care problems. The articles that were about vaccines were not negative. 

One vaccine study was funded by the Koch foundation, and found that infection was more prevalent amongst un-vaccinated children in Germany. 

In the set Matt looked through it's almost ALL about Sudden Infant Death Syndrome. Whether sleeping on your stomach causes SIDS, or whether you can identify SIDS victims by lung hemorages, or whether sleeping on sheepskin causes SIDS. One interesting study showed vaccines reduced the risk of SIDS. One showed breastfeeding reduces SIDS while another showed that breastfeeding while smoking cigarettes can increase the risk of SIDS.

There were a handful of other articles that looked at the outcomes of vaccines. Some found minor effects (red skin, swollen testicles), most reccomended vaccines and only one said current studies around vaccines are deficient, but only in convey the possible side effects.

In the set Sofia looked at seven articles actually did deal with vaccination, although not a single one had any evidence that might be considered favorable to the anti-vaccination argument. The rest dealt with either gastro-intestinal issues, SIDS, or breastfeeding



## Looking at the News 

Chiara also did an analysis of news coverage over the past month.

In [ ]:
%%sh
pip install newsapi-python

In [126]:
from newsapi import NewsApiClient
newsapi = NewsApiClient(api_key='a2099994d8494516ae2731bee998da6d')
all_articles = newsapi.get_everything(q='measles',
                                     from_param='2019-04-14',
                                     to='2019-05-14',
                                     language='en',
                                     sort_by='relevancy',
                                     page=3)
print(all_articles)

{'status': 'ok', 'totalResults': 4944, 'articles': [{'source': {'id': None, 'name': 'Health24.com'}, 'author': None, 'title': 'Worldwide measles cases have nearly tripled so far in 2019, says WHO - Health24', 'description': 'Worldwide measles cases have nearly tripled so far in 2019, says WHO Health24 Measles cases up by 300% so far in 2019, says WHO Scroll.in Measles cases triple globally in 2019, says UN BBC News Measles, Science and the Anti-Vaxxers The New York Times Measles …', 'url': 'https://www.health24.com/Medical/Childhood-diseases/Measles/worldwide-measles-cases-have-nearly-tripled-so-far-in-2019-says-who-20190416', 'urlToImage': 'http://cdn.24.co.za/files/Cms/General/d/4951/9f9e060ac0324d3d843555836bd14a8f.jpg', 'publishedAt': '2019-04-16T06:02:47Z', 'content': 'The World Health Organization (WHO) said on Monday that measles cases have continued to climb in 2019, with reported cases in the first three months of 2019 nearly 300% higher versus the same period last year.\r\nMe

In [127]:
from pprint import pprint
pprint(all_articles)

{'articles': [{'author': None,
               'content': 'The World Health Organization (WHO) said on Monday '
                          'that measles cases have continued to climb in 2019, '
                          'with reported cases in the first three months of '
                          '2019 nearly 300% higher versus the same period last '
                          'year.\r\n'
                          'Measles is a highly contagious disease, and has … '
                          '[+1045 chars]',
               'description': 'Worldwide measles cases have nearly tripled so '
                              'far in 2019, says WHO Health24 Measles cases up '
                              'by 300% so far in 2019, says WHO Scroll.in '
                              'Measles cases triple globally in 2019, says UN '
                              'BBC News Measles, Science and the Anti-Vaxxers '
                              'The New York Times Measles …',
               'publishedAt': 

In [128]:
all_articles.keys

<function dict.keys>

In [129]:
all_articles.keys()

dict_keys(['status', 'totalResults', 'articles'])

In [132]:
all_articles['articles']

[{'source': {'id': None, 'name': 'Health24.com'},
  'author': None,
  'title': 'Worldwide measles cases have nearly tripled so far in 2019, says WHO - Health24',
  'description': 'Worldwide measles cases have nearly tripled so far in 2019, says WHO Health24 Measles cases up by 300% so far in 2019, says WHO Scroll.in Measles cases triple globally in 2019, says UN BBC News Measles, Science and the Anti-Vaxxers The New York Times Measles …',
  'url': 'https://www.health24.com/Medical/Childhood-diseases/Measles/worldwide-measles-cases-have-nearly-tripled-so-far-in-2019-says-who-20190416',
  'urlToImage': 'http://cdn.24.co.za/files/Cms/General/d/4951/9f9e060ac0324d3d843555836bd14a8f.jpg',
  'publishedAt': '2019-04-16T06:02:47Z',
  'content': 'The World Health Organization (WHO) said on Monday that measles cases have continued to climb in 2019, with reported cases in the first three months of 2019 nearly 300% higher versus the same period last year.\r\nMeasles is a highly contagious disease,

Our searches revealed there have been

* 1288 articles about "anti-vaccination"
* 1288 for "anti-vaccine"
* 105 about "chicken pox"
* 105 misinformation AND measles
* 359 with misinformation AND anti-vaccination

## Looking at vaccination databases

Matt took a look at publicly available data concerning children vaccinated in New York schools

On April 2, we accessed [NYC Open Data](https://opendata.cityofnewyork.us) and found the School Immunization Survey data. These data contained the information collected from a survey of immunization records for each year since 2012-2013 from all schools in the state. 

They were broken down by percent immunized for diseases like polio and measles, and listed the percent of students who claimed religious or medical exemptions. It also contained information about the school district, like zip code, location and county. About 33,000 schools in total were in the data set. All counties except for those in NYC were removed from our data for analysis.

Public schools measle vaccine rates were never below 93%, the amount estimated to maintain "herd immunity." 

Private school did not fair as well. Of the more than 2,500 surveys in this data, 259 didn't meet the "herd immunity" standard. Some of them, like [Redeemer Lutheran School](http://rlgschool.org) were below the immunity threshold every single year on record.

[Rudolf Steiner School](https://steiner.edu) stood out in the data for a different reason. It's homepage touts that it's celebrating 90 years in buisness and claims to be "[a] tradition that ignites modern minds." [It costs](https://www.privateschoolreview.com/rudolf-steiner-school-profile) just under $50,000 per year in tuition, but only 55 percent of its students were fully immunized. For measles, Rudolf Steiner was near the lowest threshold for herd immunity.

Another school that stood apart was [Sister Clara Mohammed Elementary School](http://nycmschools.com) sitting just on the other side of Morningside Park. This is the only school to claim 100 percent religious exemption.